src/main/java/org/squiddev/cobalt/lib/BaseLib.java

```diff
@@ -337,6 +351,12 @@ protected Varargs invoke(LuaState state, DebugFrame di, Varargs args) throws Lua
 
 				case 2: // "load", // ( func|str [,chunkname[, mode[, env]]] ) -> chunk | nil, msg
 				{
+					// We'd typically rely on the argument checks in LuaValue, but those don't give us argument numbers so we do arg checks ahead of time.
```

Comment:
Just wondering why this is here? Agree that the current system is bad, but would rather handle it consistently everywhere (see #31).

Reply:
The CC:T tests fail here because they expect the error message to be `bad argument #1 (expected function or string, got <type>)`, but without the AoT type checks it errors with `bad argument (function expected, got <type>)`. I put this in to make the tests pass, but I suppose it could be removed if the tests are reworked.